testing pypi release test (#51)
mozilla-ai/lm-buddy: CONTRIBUTING.md

```
poetry publish --build
```

## Publishing  (automated)

> `.github/workflows/publish.yaml' contains the GitHub Action used to do releases and push wheels to PyPI.

Reviewer: ```suggestion
`.github/workflows/publish.yaml` contains the GitHub Action used to do releases and push wheels to PyPI.
```

Ticks were off, so was not displaying code style

> There are two subcases - draft/dev and the 'real' release.

Reviewer: ```suggestion
There are two cases - 'draft' and the 'real' releases.
```

> ### Draft / Test releases

Reviewer: ```suggestion
### Draft releases
```

> If a contributed opens a PR to `main` with a git version tag (e.g., `vX.Y.Z`) , the draft process will start which does the following:

Reviewer: ```suggestion
If a contributer opens a PR to `main` with a git version tag (e.g., `vX.Y.Z`) , the draft process will start which does the following:
```

- upload a version to test PyPI 
- make a draft release on github.

> this is to ensure the full process works before merging to main.

Reviewer: ```suggestion
This is to ensure the full process works before merging to `main`.
```

> ### Real release

Reviewer: ```suggestion
### Real releases
```

If a commit is made to `main` with a git version tag (e.g., `vX.Y.Z`), the full process will run, and upload the package to PyPI and make a formal Release. 

The workflow should look like the following:

- make your changes that will be included in a release (could be a long running dev branch or otherwise)
- update the version in `pyproject.toml`
- add an annotated tag, e.g.:
  * `git tag -a v1.4.0 -m "1.4.0 prepped for release; see ___ for changes"` or
  * `git tag -a v1.4.3 -m "1.4.3 is a patch for issue ____"`
- push your changes to upstream 
- open a PR to main
- verify that the project builds
- land PR

> 
> 
> 
> 
> 

Reviewer: ```suggestion
```
